test/rgb_matrix/animation/config_test.exs

```diff
@@ -1,52 +1,217 @@
 defmodule ConfigTest do
   use ExUnit.Case
 
+  alias RGBMatrix.Animation.Config
   alias RGBMatrix.Animation.Config.FieldType.{Integer, Option}
 
+  import ExUnit.CaptureLog
+
   defmodule MockConfig do
     @fields [
```

Comment:
We need the `@before_compile` to use a module attribute as an accumulator for fields. `@before_compile` is the step in compilation where module attribtues can be modified. By the time `use Config` is processed, the module attributes seem to be defined and fixed (I would love to be wrong).

`MockConfig` just defines the `@fields` attribute in the same way all the macro magic does, only using literals instead of macros.

Let's discuss this at some point soon and see if we can come up with a more versatile solution.

Reply:
>By the time `use Config` is processed, the module attributes seem to be defined and fixed

Ah, that's too bad. I forget if `require` behaves any differently, but my guess would be that it doesn't.